### PR TITLE
Add note about Eclipse IDE support of Angular LS

### PR DESCRIPTION
**Pull request by mickaelistria**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

I don't know how the editors are sorted, whether there is a preference to put FOSS editors above others on the list or whatnot. So I just added Eclipse IDE as last one because it's the last one to declare itself. Obviously, I wouldn't mind having it higher in the list if it's appropriate.

### CODE REVIEW

**Comment by petebacondarwin** [comment]
Does this work with the new Ivy language service?

**Comment by mickaelistria** [comment]
> Does this work with the new Ivy language service?

It's using https://github.com/eclipse/wildwebdeveloper/blob/2bf1d05c0092d0d255b705d113ab95decacac68d/org.eclipse.wildwebdeveloper/pom.xml#L156 , not exactly sure what qualifies as "Ivy" or not.

**Comment by mickaelistria** [comment]
What do I need to do for this to be good to merge?

**Comment by petebacondarwin** [comment]
It is awaiting a review from @kyliau.

I think that whether or not it supports Ivy, it is probably valid to land this as long as the project is deemed acceptable. I note that it is not an official plugin but a community effort. In that regard we should probably also get a review from @mgechev .

**Comment by mickaelistria** [comment]
>  I note that it is not an official plugin but a community effort.

What qualifies as "official" and what makes that an OSS community project cannot be "official"? It's an official Eclipse Foundation project, that is included in some flavours the official Eclipse IDE. It seems just as much official as most other entries on that same page.

**Comment by kyliau** [comment]
> Does this work with the new Ivy language service?

It uses `@angular/language-server`, so technically yes, it works with the new Ivy language service. However the client probably needs to expose an option for users to opt-in (for now). We'll make Ivy LS the default in v12.

**Review by kyliau — APPROVED** [review]

File: aio/content/guide/language-service.md [hunk 1]
```diff
@@ -103,7 +103,9 @@ npm install --save-dev @angular/language-service
 
 This allows the Angular Language Service to provide diagnostics and completions in `.ts` files.
 
+### Eclipse IDE
 
+Either directly install the "Eclipse IDE for Web and JavaScript developers" package which comes with the Angular Language Server included, or from other Eclipse IDE packages, use Help > Eclipse Marketplace to find and install [Eclipse Wild Web Developer](https://marketplace.eclipse.org/content/wild-web-developer-html-css-javascript-typescript-nodejs-angular-json-yaml-kubernetes-xml).
```
Comment (mgechev) [inline aio/content/guide/language-service.md:108]: Let us be explicit that this is a community project.

**Review by mgechev — CHANGES_REQUESTED** [review]
That's awesome! I had no idea the LS comes by default with certain setups!

I left a comment to be more specific that the team does not own this integration. Aside from that LGTM!

File: aio/content/guide/language-service.md (hunk 1 above)
Comment (mickaelistria) [reply aio/content/guide/language-service.md:108]: What do you mean by "the team"? If I look at WebStorm, there is no specific mention about anything of that sort; so I don't get what to add and why for the case of Eclipse IDE.

Comment (atscott) [reply aio/content/guide/language-service.md:108]: The Webstorm owners, Jetbrains, publish and manage the Angular extension. You can see that they are listed as the vendor under additional information. The Angular team publishes and manages the extension for VSCode. These, in effect, are "official" extensions for those respective IDEs. The Eclipse extension is published and managed by the community, not by the Eclipse team/organization.

Comment (mickaelistria) [reply aio/content/guide/language-service.md:108]: In Eclipse world, there is only 1 organization: the Eclipse Foundation does "host" projects but do not produce nor maintain software. And there is not 1 team of developers/maintainers but a variety of teams focused on subcomponents; the Eclipse IDE gathers the work of many teams, their is no such "Eclipse IDE" team. On that aspect, the team is the community, and any Eclipse.org project (including Wild Web Developer) is as official as the Ecipse IDE. Eclipse Platform or any Eclipse Foundation project.
But yes, it's a community project just like any Eclipse project. That's the mandate of the open-source way at Eclipse: a project must be an open community and mustn't be restricted to specific subgroup of individuals according to their employer.
But we're still talking about an official team of committers (OK, not bound to 1 specific vendor) officially approved by the Eclipse Foundation to maintain an official Eclipse project which is shipped as part of an official package of the Eclipse IDE (the "Eclipse IDE for JavaScript and Web developers" is a top-level artifact to download from official Eclipse IDE download side).
Community and official are not opposite terms.

Comment (mgechev) [reply aio/content/guide/language-service.md:108]: >What do you mean by "the team"? If I look at WebStorm, there is no specific mention about anything of that sort; so I don't get what to add and why for the case of Eclipse IDE.

Excuse me for the confusion. I mean the Angular team. We just want to set the correct expectations that this is not an officially maintained solution by the Angular team at Google.

Comment (mickaelistria) [reply aio/content/guide/language-service.md:108]: OK. Can you please then update the entry for WebStorm to clarify as well that it's not maintained by Angular team and I'll mimic the wording for Eclipse IDE?

Comment (mickaelistria) [reply aio/content/guide/language-service.md:108]: Maybe it's clearer to do things the other way round: on the VSCode integration, we add a note "This Visual Studio Code is officially maintained by the Angular team." and leave the other entries without specific note about who does support it.

Comment (mgechev) [reply aio/content/guide/language-service.md:108]: >OK. Can you please then update the entry for WebStorm to clarify as well that it's not maintained by Angular team and I'll mimic the wording for Eclipse IDE?

Yes, I'll open a PR for this.

**Review by mgechev — APPROVED** [review]

**Comment by mickaelistria** [comment]
Since #40647 got merged, I rebased my change on top of latest master to more easily review this change with the newer context.

**Review by atscott — APPROVED** [review]

**Comment by kyliau** [comment]
@mickaelistria I updated the commit message so that it satisfies the lint check. This should be good to go, thanks!

**Comment by mickaelistria** [comment]
Thanks for the merge and all the help before that happened!

**Pull request closed without merging**